src/lib/errorWithContext.ts

```diff
@@ -0,0 +1,12 @@
+export class ErrorWithContext extends Error {
+  constructor(message: string, public readonly context: object) {
+    super(message);
+    this.name = "ErrorWithContext";
+  }
+
+  static fromUnknown(e: unknown) {
+    return e instanceof ErrorWithContext
+      ? e
+      : new ErrorWithContext(`unknown error`, { context: JSON.stringify(e) });
```

Comment:
I don't think you can reliably do `JSON.stringify` on an Error - that's why we had that `serializeError` library.

Reply:
Ahh, yep, good catch. I moved this over from previous logic where it was checking `e instanceof Error` instead of `ErrorWithContext`. That's why `stringify` was in there. Will update to use `serializeError` here instead.